### ISSUE TITLE
False positive `import-error` with distutils.errors

### ISSUE THREAD

**Issue by akosthekiss** [pylint-dev/pylint#9238]
### Bug description

```python
import distutils.errors
```

### Command used

```shell
pylint test.py
```

### Pylint output

```python
************* Module test
test.py:1: [E0401(import-error), ] Unable to import 'distutils.errors'
```

### Expected behavior

No error is expected, since `python3 test.py` raises no errors either.

### Pylint version

```shell
pylint 3.0.2
astroid 3.0.1
Python 3.12.0 (main, Oct  5 2023, 15:44:07) [Clang 14.0.3 (clang-1403.0.22.14.1)]
```

### OS / Environment

macOS 13.6 (Darwin 22.6.0 arm64)

### Additional dependencies

```python
setuptools==68.2.2
```

**Comment by ejpenney** [comment]
I'm also seeing this same behavior with distutils.file_util:

```
[E0401(import-error), ] Unable to import 'distutils.file_util'
```

Windows 11 23H2 (OS Build 22631.4751)
```
pylint             3.3.4
astroid            3.3.8
setuptools         75.8.0
Python 3.13.1 (tags/v3.13.1:0671451, Dec  3 2024, 19:06:28) [MSC v.1942 64 bit (AMD64)]
```

This import works fine in Python, but pylint flags it as an import error (false positive).
